admin-part/src/main/kotlin/jvm/Ext.kt

```diff
@@ -24,20 +24,55 @@ import java.io.*
 
 internal fun ClassCounter.parseMethods(classBytes: ByteArray): List<Method> = run {
     val classParser = ClassParser(ByteArrayInputStream(classBytes), fullName)
-    val parsedMethods = classParser.parse().run {
+    val parsedClass = classParser.parse()
+    val parsedMethods = parsedClass.run {
         methods.associateBy { signature(fullName, it.name, it.signature) }
     }
+    val lambdas = parsedMethods.filter { "lambda" in it.key }
     methods.map { m ->
         val method = parsedMethods[m.sign.weakIntern()]
+        val parser = LambdaParser(parsedClass, lambdas)
+        val hash = parser.checksum(method)
         Method(
             ownerClass = fullName.weakIntern(),
             name = methodName(m.name, fullName),
             desc = m.desc.weakIntern(),
-            hash = method.checksum()
+            hash = hash.total,
+            lambdasHash = parser.lambdaHash.value,
+            hashWithoutLambda = hash.withoutLambda
         )
     }
 }
 
-private fun org.apache.bcel.classfile.Method?.checksum(): String = (this?.code?.run {
-    Utility.codeToString(code, constantPool, 0, length, false)
-} ?: "").crc64.weakIntern()
+private fun LambdaParser.checksum(
+    method: org.apache.bcel.classfile.Method?
+): CodeToString = (method?.code?.run {
+    codeToString(code, constantPool, 0, length, false)
+} ?: CodeToString()).run {
+    CodeToString(withoutLambda = withoutLambda.crc64.weakIntern(), total = total.crc64.weakIntern())
+}
+
+/**
+ * For each lambda creates a bootstrap method, which make references back into the main constant pool.
+ * Bootstrap methods have a fixed set of arguments, the second of them is the index of the lambda method
+ * in the constant pool.
+ * For more info: https://blogs.oracle.com/javamagazine/post/behind-the-scenes-how-do-lambda-expressions-really-work-in-java
+ */
+internal fun JavaClass.getParsedBootstrapMethods() = attributes
+    .filterIsInstance<BootstrapMethods>()
+    .firstOrNull()
+    ?.bootstrapMethods
+    ?.map { bootstrapMethod ->
+        bootstrapMethod.bootstrapArguments[1]
```

Comment:
```suggestion
        val refLamda = 1
        bootstrapMethod.bootstrapArguments[refLamba]
```
or smth another name